src/main/build/version.c

```diff
@@ -26,3 +26,9 @@ const char * const targetName = __TARGET__;
 const char * const shortGitRevision = __REVISION__;
 const char * const buildDate = __DATE__;
 const char * const buildTime = __TIME__;
+
+#ifdef BUILD_KEY
+const char * const buildKey = STR(BUILD_KEY);
+#else
+const char * const buildKey = " ";
+#endif
```

Comment:
Maybe define variable only once, but it is ugly
```suggestion
const char * const buildKey
#ifdef BUILD_KEY
 = STR(BUILD_KEY);
#else 
 = " ";
#endif
```

Why non-empty string?
And consider using NULL instead ...

Reply:
@ledvinap good point about release name

```
#ifdef BUILD_KEY
    const char * const buildKey = STR(BUILD_KEY);
#ifdef RELEASE_NAME
    const char * const releaseName = STR(RELEASE_NAME);
#else
    const char * const releaseName = NULL;
#endif
#else
    const char * const buildKey = NULL;
#endif
```

Reply:
@blckmn what will be our retention policy?

Currently we retrieve buildkey from current release only. So when upgrading to the next version we won't get details for cloud build options as the version won't match?

Reply:
> ```
> #ifdef BUILD_KEY
>     const char * const buildKey = STR(BUILD_KEY);
> #ifdef RELEASE_NAME
>     const char * const releaseName = STR(RELEASE_NAME);
> #else
>     const char * const releaseName = NULL;
> #endif
> #else
>     const char * const buildKey = NULL;
> #endif
> ```

`releaseName` won't be defined at all without BUILD_KEY

Personally, I'd define both independently

Reply:
> releaseName won't be defined at all without BUILD_KEY

Agree, but not doing so does terminate linking SITL using `make all_rev`

Reply:
> Personally, I'd define both independently

Done :+1:

Reply:
Retention will be something like 18-24 months. But we can adjust as required. It will largely depend on storage costs.